server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/AllocationUtils.java

```diff
@@ -0,0 +1,365 @@
+/*******************************************************************************
+ * Copyright (c) 2021 University of Illinois and others.  All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Mozilla Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.mozilla.org/en-US/MPL/2.0/
+ *
+ *   Contributors:
+ *   Yong Wook Kim (NCSA) - initial API and implementation
+ *******************************************************************************/
+
+package edu.illinois.ncsa.incore.common.utils;
+
+import edu.illinois.ncsa.incore.common.auth.Authorizer;
+import edu.illinois.ncsa.incore.common.exceptions.IncoreHTTPException;
+import edu.illinois.ncsa.incore.common.models.Allocation;
+import edu.illinois.ncsa.incore.common.models.Space;
+import edu.illinois.ncsa.incore.common.models.SpaceUsage;
+import edu.illinois.ncsa.incore.common.dao.IAllocationRepository;
+import edu.illinois.ncsa.incore.common.dao.ISpaceRepository;
+import edu.illinois.ncsa.incore.common.dao.MongoSpaceDBRepository;
+import edu.illinois.ncsa.incore.common.dao.MongoAllocationDBRepository;
+import org.apache.log4j.Logger;
+
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.core.Response;
+
+public class AllocationUtils {
+
+    /***
+     * This method receives a username and to see if the user's number of dataset is within the allocation.
+     *
+     * @param allocationRepository IAllocationRepository injected by controller
+     * @param spaceRepository ISpaceRepository injected by controller
+     * @param username string representation of username
+     * @return postOk boolean if the user can post more datasets
+     */
+    public static Boolean canCreateDataset(IAllocationRepository allocationRepository,
+                                           ISpaceRepository spaceRepository,
+                                           String username) {
+        // check allocation first
+        Space space = spaceRepository.getSpaceByName(username);
+        Allocation allocation = new Allocation();   // get default allocation
+        SpaceUsage usage = new SpaceUsage();
+        Boolean postOk = false;
+
+        // if space is null, it means that this post should be the very first post by the user
+        // so the check should be passed.
+        if (space == null) {
+            // First POST, no need to check the allocation
+            postOk = true;
+        } else {
+            String spaceId = space.getId();
+            usage = space.getUsage();
```

Comment:
This can be removed, there is already a call to space.getUsage on line 63.

Reply:
removed